impeller/compiler/code_gen_template.h

```diff
@@ -118,12 +118,11 @@ struct {{camel_case(shader_name)}}{{camel_case(shader_stage)}}Shader {
   // ===========================================================================
 {% for sampled_image in sampled_images %}
 
-  static constexpr auto kResource{{camel_case(sampled_image.name)}} = SampledImageSlot { // {{sampled_image.name}}
+  static constexpr auto kResource{{camel_case(sampled_image.name)}} = ShaderUniformSlot { // {{sampled_image.name}}
     "{{sampled_image.name}}",      // name
-    {{sampled_image.ext_res_0}}u,  // texture
-    {{sampled_image.ext_res_1}}u,  // sampler
-    {{sampled_image.binding}}u,    // binding
+    {{sampled_image.ext_res_0}}u,  // ext_res_0
```

Comment:
Can we have a better name for this?  This represents which texture unit this will be bound to right?  `ext_res_0` doesn't mean anything to me and seems to be a Metal-ism?

Reply:
Back to texture_index, with a note that this is ext_res_0